Formula/berkeley-db@5.rb

```diff
@@ -0,0 +1,93 @@
+class BerkeleyDbAT5 < Formula
+  desc "High performance key/value database"
+  homepage "http://www.oracle.com/technology/products/berkeley-db/index.html"
+  url "http://download.oracle.com/berkeley-db/db-5.1.19.tar.gz"
+  sha256 "0194d4ca9266ba1a1c0bfbc233b18bfd05f63163453c81ebcdfdc7112d5ac850"
+
+  keg_only :versioned_formula
+
+  # Fix build under Xcode 4.6
+  patch :DATA
```

Comment:
Doing `inreplace "src/dbinc/atomic.h", " __atomic_compare_exchange(", " __atomic_compare_exchange_db("` in `install` is probably preferable, particularly if you can guard if with `if MacOS::Xcode.installed? && MacOS::Xcode.version >= "1.2.3"` or `if ENV.compiler == :clang && DevelopmentTools.clang_build_version > 123` or similar. Thanks!